### PR TITLE
[Bugfix:System] Fix Spelling Issues

### LINKED ISSUE #8582: Add spell check github action
**What problem are you trying to solve with Submitty**
Remove spelling errors from Submitty.

**Describe the way you'd like to solve this problem**
Use codespell to spellcheck the repo at the time of commit.

### PR DESCRIPTION
**Pull request by IDzyre**
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Spell check has not been performed on the project in years. Thus, the repository has many spelling errors.

### What is the new behavior?
Related to #8582 
Codespell was run on a development computer to identify spelling errors in the repository. These are the resulting spelling fixes the development team wanted to address at this time.

### CODE REVIEW

File: more_autograding_examples/left_right_exam_seating/config/config.json
Comment (prestoncarman) [inline more_autograding_examples/left_right_exam_seating/config/config.json:4]: This should probably be "\n We"

File: more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp
Comment (prestoncarman) [inline more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp:21]: I think this may be part of the example. Instead of having a spelling error, how about having it say "years". This will still cause a compile error, but be spelled correctly.

**Review by prestoncarman — CHANGES_REQUESTED** [review]
In a few places the spelling change was in correct. At least in context it should be something else.

File: more_autograding_examples/left_right_exam_seating/config/config.json
Comment (williamjallen) [reply more_autograding_examples/left_right_exam_seating/config/config.json:4]: Doing so will add a space before "We", which may be undesired in some cases.  I think that particular part should be left alone.

**Review by williamjallen — CHANGES_REQUESTED** [review]
In general, I think a lot of these changes are promising and should definitely be merged at some point.  That said, there are also a number of locations (particularly in the tests) where the changes made are inappropriate.

As for adding codespell as a permanent GH Action, I'm somewhat divided on whether it is a good idea or not.  On one hand, we clearly have a large number of misspellings which have accumulated over the years, while on the other hand it could easily become a nuisance if it often detects false positives.  At a minimum, I think many of the changes in this PR are good and can be merged regardless of whether codespell ultimately gets added.

**Comment by prestoncarman** [comment]
@williamjallen Thanks for doing a code review. I think you have some fair points about spell checking tools.

I have worked with two different spell checking tools. The first one caused a major headache for developers. Basically each commit needed a spelling exception to be added to the ignore list. We ended up with over one hundred fifty exceptions. We eventually switched to CodeSpell (the one in this PR) and currently have no word exceptions. I have found the faults positive rate to be much less, almost none existent. It does a good job of understanding how spelling works with code. Now, it's not prefect. In this PR we had to create a few word exceptions and add files with random data to the ignore list. Overall I think this will improve the project. It also encourages developers to use complete words for variable names, which I find to be helpful when reading code.

File: more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp
Comment (williamjallen) [reply more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp:21]: This test case was deliberate and should remain as-is so it is clear how it is wrong if anyone stumbles upon it in the future.

**Comment by williamjallen** [comment]
We discussed this PR at the Submitty meeting on Wednesday.  The general consensus was that we are uncomfortable adding codespell at this time because of the likelihood of us needing to update the ignore list on  a regular basis.  Additionally, adding codespell has the potential to lull developers into a false sense of security. Codespell uses a [dictionary](https://github.com/codespell-project/codespell/tree/master/codespell_lib/data) of misspellings, not a list of valid words, which limits its effectiveness.  Several developers have also noticed instances where codespell did not catch spelling mistakes on the documentation repository.

That said, I think many of the changes in this PR are good, and can be merged soon once the tests pass and a thorough code review is completed.  This PR touches a lot of files and will likely cause merge conflicts with many existing PRs so I think it would be best to limit it to fixing misspellings and not "fixing" abbreviated variable names, to limit the amount of code this PR touches.

File: more_autograding_examples/minimal_code_editing/submissions/noise.cpp
Comment (prestoncarman) [inline more_autograding_examples/minimal_code_editing/submissions/noise.cpp:1]: This file needs to be reverted. No changes should be made.

File: more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp
Comment (prestoncarman) [reply more_autograding_examples/cpp_simple_lab/submissions/julian_compile_error.cpp:21]: Do you think a comment should be added to the file to help clarify the error?

**Review by prestoncarman — CHANGES_REQUESTED** [review]

**Comment by prestoncarman** [comment]
Thanks for considering this Github Action. I understand that not all teams and projects will want to include codespell.

@IDzyre please remove the configuration for codespell and the github action from PR.

**Review by williamjallen — CHANGES_REQUESTED** [review]
I looked at the changes in this PR in more depth and had a few more comments.  After these comments are addressed, the tests pass, and the title is edited to reflect the new purpose of this PR, this should be ready to merge after a final review.

File: migration/migrator/migrations/course/20180611125006_alphanumeric_sections.py
Comment (MasterOdin) [reply migration/migrator/migrations/course/20180611125006_alphanumeric_sections.py:2]: I disagree and this should be included. While the migration is _probably_ not ever going to be looked at again, that doesn't mean it won't, and also how much someone looks at a piece of code should not be a reason to not fix comment misspellings, unless fixing the misspelling was going to trigger some high cost for deploying or something, which it's not here.

File: migration/migrator/migrations/master/20191107184139_organize_courses_by_term.py
Comment (MasterOdin) [reply migration/migrator/migrations/master/20191107184139_organize_courses_by_term.py:24]: Same here. Fix comments in old migrations should be allowed and fine.

File: site/app/templates/admin/admin_gradeable/rubric/CheckpointsRubric.twig
Comment (IDzyre) [reply site/app/templates/admin/admin_gradeable/rubric/CheckpointsRubric.twig:21]: I believe that this button should not appear before there is a checkpoint to remove. If you add and remove a checkpoint, having one checkpoint left, the "Remove Last Checkpoint" button isn't displayed to the user. Also, I believe that the property here should be 'display:none' not 'visibility:hidden', to be consistent with the jQuery show/hide functions used.

**Review by prestoncarman — APPROVED** [review]

File: more_autograding_examples/minimal_code_editing/submissions/noise.cpp
Comment (IDzyre) [reply more_autograding_examples/minimal_code_editing/submissions/noise.cpp:24]: This is now reverting the file to the file in the current master branch. Some whitespace at the end of the file is required for the testing to pass.

Comment (IDzyre) [reply more_autograding_examples/minimal_code_editing/submissions/noise.cpp]: > This is now reverting the file to the file in the current master branch. Some whitespace at the end of the file is required for the testing to pass.

A misunderstanding, i had copied what I thought was the whole original file but it didn't copy the required end of the file. It is now completely reverted.

**Comment by prestoncarman** [comment]
@williamjallen I think all of your concerns have been addressed with this PR. Could you look at it again?

File: .setup/bin/reset_system.py
Comment (williamjallen) [inline .setup/bin/reset_system.py:100]: The `.` seems deliberate here.

File: site/app/libraries/Utils.php
Comment (williamjallen) [inline site/app/libraries/Utils.php:122]:
```suggestion
     * TODO: Make this a stronger check than just on the appended file extension to the name
```

**Review by williamjallen — CHANGES_REQUESTED** [review]
Once these changes are addressed, I will approve.

**Review by williamjallen — APPROVED** [review]

File: .setup/bin/reset_system.py
Comment (IDzyre) [reply .setup/bin/reset_system.py]: This is now reset to MOT.D

**Comment by MasterOdin** [comment]
Thanks for sticking with this @IDzyre and @prestoncarman!

**Pull request merged**
